packages/core/src/mesh/ModelMesh.ts

```diff
@@ -14,4 +58,783 @@ export class ModelMesh extends Mesh {
+      const elementCount = this._elementCount;
+      const vertices = new Float32Array(elementCount * this.vertexCount);
+      this._verticesArray = vertices;
+      this._vertexChangeFlag = ValueChanged.All;
+      this._resetVertexArrayData(vertices);
+      this._vertexBuffer = new Buffer(
+        this._engine,
+        BufferBindFlag.VertexBuffer,
+        vertices,
+        noLongerAccessible ? BufferUsage.Static : BufferUsage.Dynamic
+      );
+      this.setVertexElements(vertexElements);
+      this.setVertexBufferBinding(new VertexBufferBinding(this._vertexBuffer, elementCount * 4));
+      this._vertexSlotChanged = false;
+      this._vertexCountChanged = false;
+    } else {
+      // TODO reset value
+      const verticesArray = this._verticesArray;
+      this._resetVertexArrayData(verticesArray);
+      this._vertexBuffer!.setData(verticesArray);
+    }
+
+    if (_indicesArray) {
+      if (!this._indexBuffer) {
+        this._indexBuffer = new Buffer(this._engine, BufferBindFlag.IndexBuffer, _indicesArray);
+        this.setIndexBufferBinding(new IndexBufferBinding(this._indexBuffer, this._indicesFormat));
+      } else {
+        this._indexBuffer.setData(this._indicesArray);
+        if (this._indexBufferBinding._format !== this._indicesFormat) {
+          this.setIndexBufferBinding(new IndexBufferBinding(this._indexBuffer, this._indicesFormat));
+        }
+      }
+    } else if (this._indexBuffer) {
+      this.setIndexBufferBinding(null);
+      this._indexBuffer = null;
+    }
+
+    if (noLongerAccessible) {
+      this._releaseCache();
+    }
+  }
+
+  /**
+   * Set IndicesArray for the mesh.
+   * @param indices - The array of indices for the mesh.
+   */
+  setIndices(indices: Uint8Array | Uint16Array | Uint32Array) {
+    if (!this._accessible) {
+      throw "Not allowed to access data while accessible is false.";
+    }
+    this._indicesArray = indices;
+    if (indices === null) {
+      return;
+    }
+
+    if (indices instanceof Uint8Array) {
+      this._indicesFormat = IndexFormat.UInt8;
+    } else if (indices instanceof Uint16Array) {
+      this._indicesFormat = IndexFormat.UInt16;
+    } else if (indices instanceof Uint32Array) {
+      this._indicesFormat = IndexFormat.UInt32;
+    }
+  }
+
+  /**
+   * Get array of indices for the mesh.
+   */
+  getIndices(): Uint8Array | Uint16Array | Uint32Array {
+    return this._indicesArray;
+  }
+
+  /**
+   * Set positions array for the mesh.
+   * @param positions - The array of positions for the mesh.
+   */
+  setPositions(positions: Vector3[]): void {
+    if (!this._accessible) {
+      throw "Not allowed to access data while accessible is false.";
+    }
+
+    this._vertexSlotChanged = true;
+    const count = positions.length;
+    this._positions = positions;
+    if (this._vertexCount !== count) {
+      this._vertexCount = count;
+      this._vertexCountChanged = true;
+    }
+  }
+
+  /**
+   * Get array of positions for the mesh.
```

Comment:
write remark like:
@remarks Please call the setPositions() method after modification to ensure that the modification takes effect.

getNormals(), getColors() And so on.